core/server/master/src/main/java/alluxio/master/file/activesync/ActiveSyncManager.java

```
@@ -0,0 +1,126 @@
+        if (!UnderFileSystemUtils.isHdfs(ufsResource.get())) {
+          throw new UnsupportedOperationException("Active Syncing is only available on HDFS currently");
+        }
+        ActiveSyncer syncer = new ActiveSyncer(mFileSystemMaster, this, mMountTable, rootPath);
+
+        Future<?> future = mFileSystemMaster.getExecutorService().submit(
+            new HeartbeatThread(HeartbeatContext.MASTER_ACTIVE_SYNC,
+                syncer,
+                (int) Configuration.getMs(PropertyKey.MASTER_ACTIVE_SYNC_INTERVAL_MS)));
+        mPollerMap.put(rootPath, future);
+
+        // Add the new sync point to the filter map
+        if (mFilterMap.containsKey(rootPath)) {
+          mFilterMap.get(rootPath).add(syncPoint);
+        } else {
+          mFilterMap.put(rootPath, new CopyOnWriteArrayList<>(Arrays.asList(syncPoint)));
+        }
+        // Add to the sync point list
+        mSyncPathList.add(syncPoint);
+      }
+      return true;
+    } else {
+      return false;
+    }
+  }
+
+  public boolean removeSyncPoint(AlluxioURI syncPoint) throws InvalidPathException {
+    if (!mSyncPathList.contains(syncPoint)) {
+      return false;
+    }
+    MountTable.Resolution resolution = mMountTable.resolve(syncPoint);
+    AlluxioURI ufsUri = resolution.getUri();
+    String rootPath = ufsUri.getRootPath();
+
+    if (mFilterMap.containsKey(rootPath)) {
+      mFilterMap.get(rootPath).remove(syncPoint);
+      if (mFilterMap.get(rootPath).isEmpty()) {
+        // syncPoint removed was the last syncPoint for the rootPath
+        mFilterMap.remove(rootPath);
+        Future<?> future = mPollerMap.remove(rootPath);
+        future.cancel(true);
+      }
+      mSyncPathList.remove(syncPoint);
+    } else {
+      mSyncPathList.remove(syncPoint);
+      // We should not be in this situation
+      throw new RuntimeException(String.format("rootPath for the syncPoint %s not found", syncPoint.toString()));
```

can we throw InvalidPathException? RuntimeException seems extreme